github/actions/client.go

```diff
@@ -975,28 +975,41 @@ func (c *Client) getActionsServiceAdminConnection(ctx context.Context, rt *regis
 
 	c.logger.Info("getting Actions tenant URL and JWT", "registrationURL", req.URL.String())
 
-	resp, err := c.Do(req)
-	if err != nil {
-		return nil, err
-	}
-	defer resp.Body.Close()
+	retry := 0
+	for {
+		resp, err := c.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
 
-	if resp.StatusCode < 200 || resp.StatusCode > 299 {
-		registrationErr := fmt.Errorf("unexpected response from Actions service during registration call: %v", resp.StatusCode)
+		if resp.StatusCode >= 200 && resp.StatusCode <= 299 {
+			var actionsServiceAdminConnection *ActionsServiceAdminConnection
+			if err := json.NewDecoder(resp.Body).Decode(&actionsServiceAdminConnection); err != nil {
+				return nil, err
+			}
 
+			return actionsServiceAdminConnection, nil
```

Comment:
Can we move handling the happy path out of the loop? I think the goal of the loop should be to handle retryable errors?